### ISSUE TITLE
Rule S2053: FP when salt is greater or equal to 16

### ISSUE THREAD

**Issue by pierre-loup-tristant-sonarsource** [SonarSource/sonar-dotnet#4290]
In [eventstore](https://peach.sonarsource.com/issues?open=AXZuEvXsxqXer6FNyCKn&projects=eventstore&resolved=false&rules=csharpsquid%3AS2053) there is a FN:

```csharp
using System.Security.Cryptography;
using EventStore.Core.Authentication;

namespace EventStore.Core.Services.Transport.Http.Authentication
{
    public class Rfc2898PasswordHashAlgorithm : PasswordHashAlgorithm
    {
        private const int HashSize = 20;
        private const int SaltSize = 16;

        public override void Hash(string password, out string hash, out string salt)
        {
            var salt_ = new byte[SaltSize];
            var randomProvider = new RNGCryptoServiceProvider();
            randomProvider.GetBytes(salt_);
            var hash_ = new Rfc2898DeriveBytes(password, salt_).GetBytes(HashSize); // Compliant: Salt is random and of length 16
        }
    }
}
```

No issue should be raised because the salt is made of 16 random bytes.
